Web Crawler/src/webcrawler/HtmlParser.java

```diff
@@ -0,0 +1,128 @@
+			// contents after scripts loaded
+			webClient.getOptions().setJavaScriptEnabled(true);
+			webClient.getOptions().setCssEnabled(false);
+			webClient.getOptions().setThrowExceptionOnScriptError(false);
+			webClient.getOptions().setThrowExceptionOnFailingStatusCode(false);
+			webClient.getOptions().setTimeout(50000);
+			webClient.getOptions().setDoNotTrackEnabled(false);
+			if (!isValidUrl(url)) {
+				webClient.close();
+				return;
+			}
+			HtmlPage page = webClient.getPage(url);
+			// Wait for scripts loading
+			Thread.sleep(3000);
+			findEmailAddresses(page, domainName);
+			findValidLinks(page, "a", domainName);
+			findValidLinks(page, "div", domainName);
+			enqueue();
+			webClient.close();
+		} catch (IOException e) {
+			System.out.println("**Failure**  HTTP response fails");
+		} catch (InterruptedException e) {
+			System.out.println("**Failure**  HTTP response fails");
+		} catch (ClassCastException e) {
+			System.out.println("**Failure**  HTTP response fails");
+		}
+	}
+
+	private void enqueue() {
+		for (String link : this.links) {
+			// The last '/' is useless
+			if (link.endsWith("/")) {
+				link = link.substring(0, link.length() - 1);
+			}
+			// Remove "www." for all urls
+			if (link.contains("://www.")) {
+				link = "http://"
+						+ link.substring(link.indexOf("://www.") + 7,
+								link.length());
+			}
+			// These key words may indicate the location of email address. So
+			// put them into priorUrl Queue
+			if (link.contains("contact") || link.contains("email")
+					|| link.contains("about") || link.contains("company")
+					|| link.contains("career")) {
+				LinkQueue.addPriorUrl(link);
+			} else {
+				LinkQueue.addUnvisitedUrl(link);
+			}
+		}
+	}
+
+	private void findValidLinks(HtmlPage page, String tagName, String domainName) {
+		DomNodeList<HtmlElement> links = page.getBody().getElementsByTagName(
+				tagName);
+
+		for (DomElement link : links) {
+			String href = link.getAttribute("href");
+			// For absolute paths
+			if (href.contains(domainName) && !href.startsWith("//")) {
+				this.links.add(href);
+			}
+			// For absolute paths with "//" ahead
+			else if (href.contains(domainName) && href.startsWith("//")) {
+				this.links.add("http:" + href);
+			}
+			// For relative paths
+			else if (href.startsWith("/") && !href.startsWith("//")) {
+				this.links.add("http://" + domainName + href);
+			}
+		}
+	}
+
+	private void findEmailAddresses(HtmlPage page, String domainName) {
+		if (page == null) {
+			return;
+		}
+		String bodyText = page.getBody().asText();
+		// Split by [\t\n\x0B\f\r]
+		String[] cookedText = bodyText.split("\\s");
+		for (int i = 0; i < cookedText.length; i++) {
+			if (cookedText[i].contains("@" + domainName)) {
+				resultSet.add(cookedText[i]);
+			}
+		}
+	}
+
+	private boolean isValidUrl(String url) {
+		// Avoid any content-type except html
+		if (url.endsWith(".pdf") || url.endsWith(".rar")
```

Comment:
how would you clean this up so if we wanted to add another extension, we didn't have to make this line longer?